### PR TITLE
Lvnodcdxi 26

### PR DESCRIPTION
**Pull request by mr-redFox**
added token generate, refresh and middleware for protected routes

### CODE REVIEW

File: src/server/controllers/ApiController.js [hunk 1]
```diff
@@ -24,12 +25,33 @@ export default class ApiController {
...
+   * @param {string} userId
+   */
+  static refreshTokenHandler (req, res) {
+    const { refresh, userId } = req.body;
+
+    const findedRefreshToken = availableTokens
+      .find((token, index) => token.ref === refresh);
+
+    if (findedRefreshToken) {
+      let index = availableTokens.indexOf(findedRefreshToken);
```
Comment (tonypntsrn) [inline src/server/controllers/ApiController.js]: You can use const here, if you don’t reassign this value

File: src/server/controllers/ApiController.js [hunk 2]
```diff
@@ -24,12 +25,33 @@ export default class ApiController {
...
+      availableTokens.splice(index, 1);
+
+      return res.status(200).json({ token: generateTokens({ userId }) });
+    } else {
```
Comment (tonypntsrn) [inline src/server/controllers/ApiController.js]: No need to use else if you return smth in if statement 
This code is not reachable

File: src/server/helpers/token.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import jwt from 'jsonwebtoken';
+import crypto from 'crypto';
+import { response } from '../helpers/errorHandler';
+
+const { SECRET = '669141b534e06cf5c0271a389b79ebc11910f86bd210a74b6f00fcb974010809' } = process.env;
```
Comment (tonypntsrn) [inline src/server/helpers/token.js]: Don’t use smth secret, if it can be seen 🙂

File: src/server/helpers/token.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  const access = jwt.sign({
+    ...payload,
+    _refresh: refresh
+  }, SECRET, { expiresIn: life });
+
+  return { refresh, access };
+};
+
+export const middleware = (req, res, next) => {
+  let token = req.headers['x-access-token'];
```
Comment (tonypntsrn) [inline src/server/helpers/token.js]: You can use const here

File: src/server/controllers/ApiController.js (hunk 2 above)
Comment (mr-redFox) [reply src/server/controllers/ApiController.js]: done

File: src/server/helpers/token.js (hunk 3 above)
Comment (mr-redFox) [reply src/server/helpers/token.js]: done

File: src/server/helpers/token.js (hunk 4 above)
Comment (mr-redFox) [reply src/server/helpers/token.js]: done

File: src/server/helpers/token.js (hunk 3 above)
Comment (mr-redFox) [reply src/server/helpers/token.js]: i used for it dotenv package, and all future env consts will in .env

File: src/server/controllers/ApiController.js (hunk 1 above)
Comment (mr-redFox) [reply src/server/controllers/ApiController.js]: done

File: src/server/helpers/token.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import crypto from 'crypto'
+import { response } from '../helpers/errorHandler'
+import dotenv from 'dotenv'
+
+dotenv.config()
+
+const { SECRET } = process.env
+const tokenLife = '15m'
+
+export const generate = (payload, life = tokenLife) => {
```
Comment (tonypntsrn) [inline src/server/helpers/token.js]: Better to use more informative names, for eg, generate what?
Refresh what?
Access what or access to??

File: src/server/.eslintrc.js [hunk 6]
```diff
@@ -1,18 +1,16 @@
...
-		es6: true,
-		node: true
+	'env': {
+	  'es6': true
 	},
-	extends: 'standard',
-	globals: {
-		Atomics: 'readonly',
-		SharedArrayBuffer: 'readonly'
+	'extends': 'standard',
```
Comment (StanislavFedyna) [inline src/server/.eslintrc.js]: Fix problem with eslint
include Airbnb, please

File: src/server/controllers/ApiController.js [hunk 7]
```diff
@@ -1,42 +1,70 @@
...
+  /**
+   *
+   * @param {string} refresh
+   * @param {string} userId
+   */
+  static refreshTokenHandler (req, res) {
+    const { refresh, userId } = req.body
+
+    const findedRefreshToken = availableTokens
+      .find((token, index) => token.ref === refresh)
```
Comment (StanislavFedyna) [inline src/server/controllers/ApiController.js]: You can remove index param from this place if you don't used

**Review by StanislavFedyna — CHANGES_REQUESTED** [review]

File: src/server/controllers/ApiController.js (hunk 7 above)
Comment (mr-redFox) [reply src/server/controllers/ApiController.js]: added

Comment (mr-redFox) [reply src/server/controllers/ApiController.js]: fixed

File: src/server/.eslintrc.js (hunk 6 above)
Comment (mr-redFox) [reply src/server/.eslintrc.js]: airbnb linter added to server

File: src/server/helpers/token.js (hunk 5 above)
Comment (mr-redFox) [reply src/server/helpers/token.js]: renamade

**Review by StanislavFedyna — APPROVED** [review]

**Review by OstapDutkevych — APPROVED** [review]

**Review by tonypntsrn — APPROVED** [review]

**Pull request merged**